src/collection.js

```diff
@@ -20,15 +25,47 @@ CKEDITOR.define( [ 'emittermixin', 'ckeditorerror', 'utils' ], function( Emitter
 		 * Creates a new Collection instance.
 		 *
 		 * @constructor
+		 * @param {Iterale} [items] Items to be added to the collection.
+		 * @param {String} [idName='id'] The name of the property which is considered to identify an item.
 		 */
-		constructor() {
+		constructor( items, idName ) {
```

Comment:
If I could add my bits to this, I dislike a lot configuration parameters. As for methods, I think we should start getting used to have one and only `options` parameter that can be extended as needed.

In the case of a class constructor though, we could have a more elegant solution, like a configuration property. I see that it has been decided to pass the initial items to the constructor though, so we need to know it beforehand. Therefore I would still propose to call it `options`.